### ISSUE TITLE
Some good sites for testing.

### ISSUE THREAD

**Issue by narcolepticinsomniac** [schomery/popup-blocker#3]
Hey, I emailed you the other day. The whitelist is a big improvement. I've been trying to use your extension exclusively, and while it catches a couple the best ones missed, it also misses a ton of popups.

For testing purposes, no sites are quite as in love with popups as the sleazy filehosters. Click on the links on this page, for instance (have some good adblocking in place first): 

http://www.primewire.ag/watch-2780265-Jason-Bourne

The best all around popup blocker for Chrome is currently Poper Blocker, which catches at least 90%, but it was sold recently and is less trustworthy imo. I hope yours improves to the point I can ditch it.

**Comment by schomery** [comment]
> The whitelist is a big improvement. 

The new dev version has two types of filters. 1. Whitelist the entire tab, 2. Whitelist certain popups based on their URLs

> http://www.primewire.ag/watch-2780265-Jason-Bourne

So even if I disable my adblocker in this website I don't see any popup being passed. Can you point me to what is missed?

**Comment by narcolepticinsomniac** [comment]
Yeah, the "third-party" whitelist is a pretty sweet idea. My fault, I guess I wasn't clear. The page I linked is full of links to the filehosters I was referring to. While navigating to get the videos to play, most have unblocked popups along the way.

**Comment by schomery** [comment]
Yah, I actually followed a few videos links;
1. I turned my adblocker off to make sure it is not blocking extra popups
2. Navigated to http://www.primewire.ag/watch-2780265-Jason-Bourne
3. Clicked on the "Version 1" link in the middle of the page
![screen shot 2016-11-18 at 10 06 49](https://cloud.githubusercontent.com/assets/11704051/20421247/f3c35a9a-ad77-11e6-8a3d-c9e5253de550.png)
4. Got two popup requests
5. I am going to allow one of these by pressing the "background" button ("background" and "redirect" buttons only open the link in a background tab, while "allow" opens the popup as it was requested by the page)
![screen shot 2016-11-18 at 10 17 57](https://cloud.githubusercontent.com/assets/11704051/20421313/4cc1bf88-ad78-11e6-993f-d893d8abdda9.png)
6. No luck with this! Going with the second one,
![screen shot 2016-11-18 at 10 19 28](https://cloud.githubusercontent.com/assets/11704051/20421343/8407bc4a-ad78-11e6-9b98-4e53b4d6a997.png)
7. Seems to be fine

Am I missing anything here?

**Comment by narcolepticinsomniac** [comment]
No, I guess I was still too vague. The first link is always a dummy ad link. I forgot because I block them. Here's some direct links from that page:

(missed altogether)
https://estream.to/mss86x8jqeot.html
http://vidtodo.com/03pbhekfsihm
https://thevideobee.to/35qlbo60f2po.html
http://vshare.eu/cw8ajp3lt68j.htm

(catches 1st, video autoplays, click anywhere and it redirects)
http://thevideo.me/74257aar9ymg
http://vidup.me/tdfxt6kp45i3

(missed by extension caught by browser)
http://www.nowvideo.sx/video/3333c2b0a6b23
http://vidzi.tv/y1o4zjusoqbn.html
http://vidbull.com/qc0itm2dotoa.html
http://www.auroravid.to/video/06fb871e3a7ff

All tested with Ublock enabled.

**Comment by schomery** [comment]
Thanks! on 0.1.8 we are covering 3 more new popup types that hopefully eliminates most of these annoying popups. Please wait for an hour to get 0.1.8 and report pages that this new version is still cannot block popups on.

p.s: I know one particular page, but it only displays popups when my adblocker is disabled. When I disable mine, Chrome is not responding at all as the page is super heavy!  Apparently there must be another creative way to issue popups that this extension is not yet covering!

**Comment by narcolepticinsomniac** [comment]
Improved, for sure. 

Fixed:

https://estream.to/mss86x8jqeot.html

http://www.auroravid.to/video/06fb871e3a7ff

Not fixed (for testing purposes on all sites, clear history/cache each time since some are kind enough not to deploy them consecutively):

[first click] http://vidtodo.com/03pbhekfsihm

[click on player] https://thevideobee.to/35qlbo60f2po.html

[first click & click on player] http://vshare.eu/cw8ajp3lt68j.htm

These sites are horrible, and aren't completely fixed with any combo of blockers. With just yours, it is a continuous loop of the landing page. Popper Stopper Pro will get you to the video, but any click on the page is an in-page redirect. The code for these is so f'd up, I don't even know if they could be fixed.

http://thevideo.me/74257aar9ymg

http://vidup.me/tdfxt6kp45i3

Somewhat buggy:

[Mostly blocked, but occasionally pops up a duplicate of the landing page you're on] http://www.nowvideo.sx/video/3333c2b0a6b23 

[Registers as blocked, but still opens and caught by browser] http://vidzi.tv/y1o4zjusoqbn.html


Like I said, no single blocker works well universally in Chrome. I've been using a combo of Poper Blocker and Popper Stopper Pro. Neither is perfect, but very few get by both. If you want to take a look at their code, here's their .crx's:

[Downloads.zip](https://github.com/schomery/popup-blocker/files/600988/Downloads.zip)

As long as you're actively developing, I'll help by testing/reporting if it's helpful. If not, just ignore me and I'll take the hint.

**Comment by schomery** [comment]
@narcolepticinsomniac great resources! there are many creative ways to generate popups. I've managed to improve the extension a bit more. Please give the latest dev a try and report back with links that can still pass the blocker.

To try the dev version:

1. Download https://github.com/schomery/popup-blocker/archive/master.zip and unzip it somewhere locally
2. Open chrome://extensions
3. Disable all the current blockers
4. Check "Developer mode" and point the "Load unpacked extension" to the root of unzipped folder

**Comment by narcolepticinsomniac** [comment]
Hey. It's "detection" rate has improved considerably, unfortunately it isn't blocking the ones it's detecting. You can use the previous example sites since none are really "fixed". It is important to clear history/cache before loading pages for testing. Clicking all the way through till the actual video plays still results in popups for all the example sites, but when you close the popup you can see that the extension detected it because your "options" balloon is present. It's detecting them, but failing to prevent them.

On a sidenote, I also get a false positive on Youtube. When a video ends, if you click the play button in the middle of the player to start the next video, it results in a false positive.

**Comment by schomery** [comment]
@narcolepticinsomniac please give https://github.com/schomery/popup-blocker/archive/master.zip a try one more time. I'm not getting any popups in this build anymore in a clean profile.

**Comment by narcolepticinsomniac** [comment]
Huh. We're having wildly different experiences then. I  made a screen record of the latest master in a clean profile with only Ublock installed alongside it, just so you don't think I'm crazy:
[2016-11-20_16-10-56.zip](https://github.com/schomery/popup-blocker/files/602343/2016-11-20_16-10-56.zip)

**Comment by schomery** [comment]
It seems like the popup blocker is not being installed properly. I am on a Mac, however, I just tested the latest dev version on a Windows machine and it worked just fine (had to press the refresh button in the chrome://extensions tab once for the blocker to be effective though).

To make sure the extension is properly installed, please drop '/test/popup-blocker.html' file into your Chrome browser and press each single test. The latest version should be able to pass all the tests. Now try one of the sites one more time and see if still popups are missed or not.

**Comment by narcolepticinsomniac** [comment]
I don't know, man. It seems to be installed fine. I side-load extensions all the time. Here's a screen record of the test, and then the same couple sites:
[popup-test.zip](https://github.com/schomery/popup-blocker/files/603430/popup-test.zip)

**Comment by schomery** [comment]
That's weird! It was with the latest dev version, right?

**Comment by narcolepticinsomniac** [comment]
Yup, latest dev. I don't get it either. The 2 sites in the vid are:

[first click] http://vidtodo.com/03pbhekfsihm

[click on player] https://thevideobee.to/35qlbo60f2po.html

With a clean cache, all popups are blocked for you? Anyway, I gotta go work for a bit, but if you have any bright ideas, I'll be happy to test whatever later on today.

**Comment by schomery** [comment]
Yes, take a look,

[Untitled-2.zip](https://github.com/schomery/popup-blocker/files/603586/Untitled-2.zip)

This is on a a clean profile.

**Comment by schomery** [comment]
Just a note; all popup blockers are working by overwriting some internal functions like `window.open` for instance. At least in this extension we do not allow anybody else to take control over this function. So if for instance another popup blocker wants to do the same, there would be a competition and the result would be unknown (at least this extension would throw an error. you can check this on the test page). So I would recommend to use one popup blocker.

**Comment by narcolepticinsomniac** [comment]
I wouldn't test yours with other popup-blockers enabled, but the problem is a conflict with another extension. With ABP or Ublock enabled, your extension doesn't work right. It didn't occur to me because I've never seen adblock interfere with a popup-blocker, but it was the only other extension I always had enabled so I tested without one and it works on the test sites. 

It doesn't matter which one was enabled, ABP and Ublock both interfered. Hopefully this can be fixed because most people who use popup-blockers also use some form of adblock. I wonder why yours has a conflict and others don't, maybe the script injection method? 

Here's the screen record of the test I ran:
[pop-test.zip](https://github.com/schomery/popup-blocker/files/604700/pop-test.zip)

**Comment by schomery** [comment]
this is the tough one! Both extensions want to take control over `window.open` and I have no idea which one is going to take the control first. I guess if you refresh the popup-blocker extension after uBlock or AdBlock Plus is installed, this extension is going to install the first listener or vice versa!

I am going to contact the developers of these popular extensions and see if we can come up with a solution.

**Comment by schomery** [comment]
I think I got this new adblock issue fixed. Please wait for a couple of hours until 0.2.1 is released.

Please let me know if are websites that still can pass 0.2.1.

**Comment by narcolepticinsomniac** [comment]
Nice work! The other blockers have issues with some popups where even when they're blocked the code keeps running, so they're blocked over and over, and sometimes functionality like searching is broken. Yours has no issues with them. It's really kicking ass so far in limited testing. I'm going to use it exclusively from now on, and I'll let you know if it misses any.

I still think being able to access the options (allow/deny/etc) via a dropdown instead of the mandatory popup bubble is a necessity. For the time-being I made a global style to hide the iframe:

iframe[src="chrome-extension://aefkmifgmaafnojlojpnekbpbmjiiogg/data/ui/ui.html"]{display:none!important;}

Also, maybe an easy "whitelist current site" button in the dropdown like some blockers have.

Really big improvements though, keep up the good work!

**Comment by schomery** [comment]
Great!

UI definitely can be improved, like we can have whitelist button for each popup domain in the UI, a way to easily whitelist the current site, or have different modes like dropdown mode that you have suggested. I'll gradually improve the UI.

I am closing this issue. Please open a new one for UI changes.

**Comment by schomery** [comment]
@narcolepticinsomniac would you please give the latest dev a try for a while. To make the extension compatible with some famous add-ons like TamperMonkey I had to allow some objects to be writable. I don't think there is still a way to bypass the blocker, but it would be nice to find it before releasing the new version.
